libpod/runtime_img.go

```diff
@@ -96,30 +173,35 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 
 	// PullImage copies the image from the source to the destination
 	var (
-		images []string
+		images       []string
+		fromRegistry bool
 	)
 
-	if signaturePolicyPath == "" {
-		signaturePolicyPath = r.config.SignaturePolicyPath
+	signaturePolicyPath := r.config.SignaturePolicyPath
+	if options.SignaturePolicyPath != "" {
+		signaturePolicyPath = options.SignaturePolicyPath
 	}
 
-	sc := common.GetSystemContext(signaturePolicyPath, "")
+	sc := common.GetSystemContext(signaturePolicyPath, options.AuthFile)
 
 	srcRef, err := alltransports.ParseImageName(imgName)
 	if err != nil {
-		defaultName := DefaultRegistry + imgName
-		srcRef2, err2 := alltransports.ParseImageName(defaultName)
-		if err2 != nil {
-			return errors.Errorf("error parsing image name %q: %v", defaultName, err2)
+		// could be trying to pull from registry with short name
+		images, err = getRegistriesToTry(imgName)
+		if err != nil {
+			fmt.Println("error getting default registries to try:", err)
+		}
+		if len(images) == 0 {
+			images = append(images, DefaultRegistry+imgName)
```

Comment:
`getRegistriesToTry` has already tried to parse (slightly mutilated) `"docker://"+imgName` if `imgName` contains a host name, or `["docker://"+hostname+imgName for hostname in sysregistries]` if it doesn’t.

So, this makes a difference only in 1) keeping the digest suffix, and 2) trying the default normalization, i.e. `"docker://docker.io"+imgName`). I assume 1) can be fixed in `getRegistriesToTry`, and then 2), if it is intentional, would be much cleaner by appending "docker.io" to `searchRegistries` in `getRegistriesToTry`.

Reply:
fixed I believe.